File: VB/WebSolution.Module.Web/CollapseLargeEditorDetailViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Editors
Imports System.Web.UI
Imports DevExpress.Web.ASPxNavBar

Namespace WebSolution.Module.Web
	Public Class CollapseLargeEditorDetailViewController
		Inherits ViewController
		Public Sub New()
			TargetViewType = ViewType.DetailView
		End Sub
		Protected Overrides Sub OnViewControlsCreated()
			MyBase.OnViewControlsCreated()
			Dim dv As DetailView = CType(View, DetailView)
			For Each item As DetailViewItem In dv.Items
				Dim control As Control = CType(item.Control, Control)
				Dim parent As Control = control.Parent
				If parent IsNot Nothing Then
					If TypeOf item Is ListPropertyEditor AndAlso dv.ViewEditMode = ViewEditMode.View Then
						Dim navbar As New ASPxNavBar()
						Dim index As Integer = parent.Controls.IndexOf(control)
						parent.Controls.RemoveAt(index)
						parent.Controls.AddAt(index, navbar)
						Dim group As New NavBarGroup(String.Empty)
						group.ExpandButtonPosition = ExpandButtonPosition.Left
						group.ContentTemplate = New NavBarGroupContentTemplate(control)
						navbar.Groups.Add(group)
					End If
				End If
			Next item
		End Sub
	End Class
	Public Class NavBarGroupContentTemplate
		Implements ITemplate
		Public Sub New(ByVal templateControl As Control)
			MyBase.New()
			templateControlCore = templateControl
		End Sub
		Private templateControlCore As Control = Nothing
		Public ReadOnly Property TemplateControl() As Control
			Get
				Return templateControlCore
			End Get
		End Property
		Public Sub InstantiateIn(ByVal container As Control) Implements ITemplate.InstantiateIn
			If TemplateControl IsNot Nothing Then
				container.Controls.Add(TemplateControl)
			End If
		End Sub
	End Class
End Namespace

File: CS/WebSolution.Module.Web/CollapseLargeEditorDetailViewController.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Editors;
using System.Web.UI;
using DevExpress.Web.ASPxNavBar;

namespace WebSolution.Module.Web {
    public class CollapseLargeEditorDetailViewController : ViewController {
        public CollapseLargeEditorDetailViewController() {
            TargetViewType = ViewType.DetailView;
        }
        protected override void OnViewControlsCreated() {
            base.OnViewControlsCreated();
            DetailView dv = (DetailView)View;
            foreach (ViewItem item in dv.Items) {
                Control control = (Control)item.Control;
                Control parent = control.Parent;
                if (parent != null) {
                    if (item is ListPropertyEditor && dv.ViewEditMode == ViewEditMode.View) {
                        ASPxNavBar navbar = new ASPxNavBar();
                        int index = parent.Controls.IndexOf(control);
                        parent.Controls.RemoveAt(index);
                        parent.Controls.AddAt(index, navbar);
                        NavBarGroup group = new NavBarGroup(string.Empty);
                        group.ExpandButtonPosition = ExpandButtonPosition.Left;
                        group.ContentTemplate = new NavBarGroupContentTemplate(control);
                        navbar.Groups.Add(group);
                    }
                }
            }
        }
    }
    public class NavBarGroupContentTemplate : ITemplate {
        public NavBarGroupContentTemplate(Control templateControl)
            : base() {
            templateControlCore = templateControl;
        }
        private Control templateControlCore = null;
        public Control TemplateControl {
            get {
                return templateControlCore;
            }
        }
        public void InstantiateIn(Control container) {
            if (TemplateControl != null) {
                container.Controls.Add(TemplateControl);
            }
        }
    }
}